cardano-sync/src/Cardano/Sync.hs

```diff
@@ -387,12 +387,20 @@ chainSyncClient dataLayer trce env queryVar metrics latestPoints currentTip acti
       ClientStNext
         { recvMsgRollForward = \blk tip ->
               logException trce "recvMsgRollForward: " $ do
-                Gauge.set (withOrigin 0 (fromIntegral . unBlockNo) (getTipBlockNo tip)) (mNodeHeight metrics)
+
+                -- Setting the value from network/node
+                let setNodeBlockHeight = mlSetNodeBlockHeight metricsLayer
+                setNodeBlockHeight (withOrigin 0 unBlockNo (getTipBlockNo tip))
+
                 details <- getSlotDetails trce env queryVar (cardanoBlockSlotNo blk)
                 newSize <- atomically $ do
                                 writeDbActionQueue actionQueue $ mkDbApply blk details
                                 lengthDbActionQueue actionQueue
-                Gauge.set (fromIntegral newSize) $ mQueuePostWrite metrics
+
+                -- The action queue size
+                let setQueuePostWrite = mlSetQueuePostWrite metricsLayer
```

Comment:
Please pull out `setQueuePostWrite` , `setNodeBlockHeight` as top level functions.

Reply:
Not sure what you mean by `top level functions`?

Reply:
Fixed.